lib/networkx/graph.rb

```diff
@@ -117,5 +117,216 @@ def remove_nodes(nodes)
                              "received #{nodes.class.name} instead."
       end
     end
+
+    # Removes edge from the graph
+    #
+    # @example
+    #   graph.remove_edge('Noida', 'Bangalore')
+    #
+    # @param node_1 [Object] the first node of the edge
+    # @param node_2 [Object] the second node of the edge
+    def remove_edge(node_1, node_2)
+      raise KeyError, "#{node_1} is not a valid node." unless @nodes.key?(node_1)
+      raise KeyError, "#{node_2} is not a valid node" unless @nodes.key?(node_2)
+      raise KeyError, 'The given edge is not a valid one.' unless @adj[node_1].key?(node_2)
+      @adj[node_1].delete(node_2)
+      @adj[node_2].delete(node_1) if node_1 != node_2
+    end
+
+    # Removes multiple edges from the graph
+    #
+    # @example
+    #   graph.remove_edges([%w[Noida Bangalore], %w[Bangalore Chennai]])
+    #
+    # @param edges [Array<Object>] the array of edges to be removed
+    def remove_edges(edges)
+      case edges
+      when Array, Set
+        edges.each { |node_1, node_2| remove_edge(node_1, node_2) }
+      else
+        raise ArgumentError, 'Expected Arguement to be Array or Set of edges, '\
+                             "received #{edges.class.name} instead."
+      end
+    end
+
+    # Adds weighted edge
+    #
+    # @example
+    #   graph.add_weighted_edge('Noida', 'Bangalore', 1000)
+    #
+    # @param node_1 [Object] the first node of the edge
+    # @param node_2 [Object] the second node of the edge
+    # @param weight [Integer] the weight value
+    def add_weighted_edge(node_1, node_2, weight)
+      add_edge(node_1, node_2, weight: weight)
+    end
+
+    # Adds multiple weighted edges
+    #
+    # @example
+    #   graph.add_weighted_edges([['Noida', 'Bangalore', 1000],
+    #                             ['Noida', 'Nagpur', 1000]])
+    #
+    # @param edges [Array<Object, Object>] the array of edges
+    # @param weight [Array<Integer>] the array of weights
+    def add_weighted_edges(edges, weights)
```

Comment:
The args provided in the documentation and the args of the method don't match. Have a look at this again.